src/opnsense/mvc/app/models/OPNsense/IPsec/Swanctl.xml

```diff
@@ -328,6 +328,7 @@
                     <NetMaskRequired>Y</NetMaskRequired>
                     <ValidationMessage>Please specify a valid CIDR subnet.</ValidationMessage>
                 </addrs>
+                <dns type=".\PoolsDNSField"/>
```

Comment:
@Monviech The `NetworkField` type (https://docs.opnsense.org/development/frontend/models_fieldtypes.html#networkfield) with some options set should be enough in this case, I don't think we need anything special here, other than that, this should indeed do the trick.

Reply:
Thank you for your feedback, my solution was way too complicated. xD
I added the NetworkField instead with the minimum of options needed to get the same result as before.

Reply:
@Monviech yep, this should be all indeed. thanks!